### DISCUSSION TITLE
Extracting session duration

### DISCUSSION THREAD

**Discussion by twheatcroft07** [dandi/helpdesk#143] [Q&A]
I am interested in understanding causes of variability in good unit number (relative to total unit number, or numbers of channels). I have been using data such as [this ](https://neurosift.app/?p=/nwb&url=https://api.dandiarchive.org/api/assets/e1e626c9-f11e-4316-95b3-44c4f485d3ef/download/&dandisetId=000363&dandisetVersion=0.230822.0128). 

Is it possible to extract the duration of the extracellular electrophysiological recording? I can see the session start time, but not the end, and I am unsure whether this represents the duration of any acquisition (e.g. of cameras) or of the ephys specifically.

Thanks!

**Comment by CodyCBakerPhD** [comment]
There is no official stop time in NWB; the 'length' of the session can be inferred by iterating over all the main data streams and estimating it based on the maximum `timestamp[-1]` (last index of time vector) or alternatively `time_series.shape[0] * rate` (since the first axis of any series should be time) over all the temporal objects

**Reply by twheatcroft07** [reply]
Perfect, thanks!

**Comment by CodyCBakerPhD** [comment]
However I caution that for chronic studies, there may be many breaks in between the start and end of what constitutes a 'session', usually annotated by the `epochs` table
